### PR TITLE
Refactored functions logs functions

### PR DESCRIPTION
**Pull request by lo9i**
Removed duplicated code and fixed the log message

### CODE REVIEW

File: src/core/default/backend.py [hunk 1]
```diff
@@ -278,25 +275,24 @@ def _delete_component(self, resource_state_uuid: str, component_name: str) -> No
...
 
-        deleteing_component = next(
-            x for x in resource_state.components if x.name == component_name
-        )
-
-        if not len(deleteing_component.resources) == 0:
+        if len(deleteing_component.resources) != 0:
             raise ComponentNotEmpty(
-                f"Can not delete Component {component_name} in Resource State {resource_state_uuid} because the component is not empty"
+                f"Can not delete Component {component_name} in Resource State {resource_state_uuid} because the it is not empty"
```
Comment (DanielSanchez97) [inline src/core/default/backend.py:292]: change to 'because it is not empty' or leave as original

**Review by DanielSanchez97 — CHANGES_REQUESTED** [review]
This PR has a lot of things going on. Can you separate it into different smaller ones to make it easier to see what each change if for.

**Review by DanielSanchez97 — CHANGES_REQUESTED** [review]
See my one note. Besides that it looks good to me.

File: src/core/default/commands/function/logs.py [hunk 2]
```diff
@@ -58,181 +57,211 @@ def add_arguments(self, parser: ArgumentParser) -> None:
...
+                cloud_client,
+                cloud_group_name,
+                stream_name,
+                limit_val,
+                start_from_head,
+                next_token_property_name)
+
+    def _paginate_stream_log_events(
+        self,
+        cloud_client,
```
[the remark this answers is not in the document]
Comment (lo9i) [reply src/core/default/commands/function/logs.py:117]: I'll take care of this right away

**Review by DanielSanchez97 — APPROVED** [review]
LGTM

**Pull request merged**